storage/innobase/include/ut0pool.h

```diff
@@ -160,6 +168,8 @@ struct Pool {
 
 		ut_ad(Factory::debug(&elem->m_type));
 
+		// UNIV_MEM_FREE(elem, sizeof(*elem));
```

Comment:
@dr-m Try uncomment this line and re-run `innodb` suite.

Reply:
@kevgs I think that the transaction pool in InnoDB is intentionally avoiding the initialization of some fields. See the macros `assert_trx_is_free()` and `assert_trx_is_inactive()` and the functions `trx_validate_state_before_free()` and `trx_create_low()`.

Because Valgrind does not appear to support methods for manipulating the A bits independently of the V bits (similar to the AddressSanitizer poison/unpoision operation), it seems that we will only be able to use the ASAN poison/unpoison in `Pool`, in the member functions `put()`, `get()` and `init()`.

Reply:
@kevgs Please submit a revised pull request against 5.5, to instrument `mem_heap_t`. When it comes to `Pool`, I have attached a patch to [MDEV-15030](https://jira.mariadb.org/browse/MDEV-15030).